internal/rukpak/convert/registryv1.go

```diff
@@ -101,9 +103,54 @@ func RegistryV1ToHelmChart(ctx context.Context, rv1 fs.FS, installNamespace stri
 		return nil, err
 	}
 
+	if err := copyMetadataPropertiesToCSV(&reg.CSV, rv1); err != nil {
+		return nil, err
+	}
+
 	return toChart(reg, installNamespace, watchNamespaces)
 }
 
+// copyMetadataPropertiesToCSV copies properties from `metadata/propeties.yaml` (in the filesystem fsys) into
+// the CSV's `.metadata.annotations['olm.properties']` value, preserving any properties that are already
+// present in the annotations.
+func copyMetadataPropertiesToCSV(csv *v1alpha1.ClusterServiceVersion, fsys fs.FS) error {
+	var allProperties []property.Property
+
+	// First load existing properties from the CSV. We want to preserve these.
+	if csvPropertiesJSON, ok := csv.Annotations["olm.properties"]; ok {
+		var csvProperties []property.Property
+		if err := json.Unmarshal([]byte(csvPropertiesJSON), &csvProperties); err != nil {
+			return fmt.Errorf("failed to unmarshal csv.metadata.annotations['olm.properties']: %w", err)
+		}
+		allProperties = append(allProperties, csvProperties...)
+	}
+
+	// Next, load properties from the metadata/properties.yaml file, if it exists.
+	metadataPropertiesJSON, err := fs.ReadFile(fsys, filepath.Join("metadata", "properties.yaml"))
+	if err != nil && !errors.Is(err, fs.ErrNotExist) {
+		return fmt.Errorf("failed to read properties.yaml file: %w", err)
+	}
+	if len(metadataPropertiesJSON) > 0 {
```

Comment:
If there are no metadataPropertiesJSON, then you can early exit this function

Reply:
That would also decrease the overall indentation of the function...

Reply:
[61cf3b30...14266380](https://github.com/operator-framework/operator-controller/compare/61cf3b30702d5ad82c48b93c74c76625f4c3d9d2..142663800809cbdbe95abe95e64c449b544fec7d)